src/macros.php

```diff
@@ -297,3 +299,49 @@
         return $this->reverse()->after($currentItem, $fallback);
     });
 }
+
+if (! Collection::hasMacro('paginate')) {
+    /*
+     * Paginate the given collection
+     *
+     * @param int $perPage
+     * @param int $total
+     * @param int $page
+     * @param string $pageName
+     *
+     * @return \Illuminate\Pagination\LengthAwarePaginator
+     */
+    Collection::macro('paginate', function (int $perPage = 15, string $pageName = 'page', int $page = null, int $total = null, array $options = []): LengthAwarePaginator {
```

Comment:
I was hesitating  how to do it, I thought it would be better if I do it like we have in `\Illuminate\Database\Eloquent\Builder::paginate()`
so for example you do `\App\Users::paginate(20)`, and for collections you can do `$yourCollection->pagiinate(20)`
The problem with my solution is that there is no columns in collection (but I can do that it will grab only certain "column" in collection) and options are not available.
Do you get what I mean?
Do I still need to do it your way?